### PR TITLE
CIV-9124 Finalise trial start screen

### PR DESCRIPTION
**Pull request by bhagyashreesharma90**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-9124

### Change description ###
Added start screen for trial arrangements
image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/common/form/models/claimSummarySection.ts [hunk 1]
```diff
@@ -54,5 +54,6 @@ export enum ClaimSummaryType {
   INPUT_ARRAY_ERROR = 'inputArrayError',
   DATE_ARRAY = 'dateArray',
   UPLOAD_ARRAY = 'uploadArray',
-  SELECT = 'select'
+  SELECT = 'select',
+  LEAD_PARAGRAPH_WITH_NO_BOTTOM_MARGIN = 'LEAD_PARAGRAPH_WITH_NO_BOTTOM_MARGIN'
```
Comment (hmcts-version1-ignacio) [inline src/main/common/form/models/claimSummarySection.ts:58]: Following the way the other components are listed, shouldn't it be better to change this to:
LEAD_PARAGRAPH_WITH_NO_BOTTOM_MARGIN = 'leadParagraphWithNoBottomMargin' ?

Comment (bhagyashreesharma90) [reply src/main/common/form/models/claimSummarySection.ts]: > Following the way the other components are listed, shouldn't it be better to change this to: LEAD_PARAGRAPH_WITH_NO_BOTTOM_MARGIN = 'leadParagraphWithNoBottomMargin' ?

done

File: src/main/routes/features/caseProgression/trialArrangements/finaliseTrialArrangementsController.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+const finalizeTrialArrangementsViewPath = 'features/caseProgression/trialArrangements/finalise-trial-arrangements';
+const finaliseTrialArrangementsController = Router();
+const civilServiceApiBaseUrl = config.get<string>('services.civilService.url');
+const civilServiceClient: CivilServiceClient = new CivilServiceClient(civilServiceApiBaseUrl);
+
+finaliseTrialArrangementsController.get([CP_FINALISE_TRIAL_ARRANGEMENTS_URL], (async (req, res, next: NextFunction) => {
+  try {
+    const claimId = req.params.id;
+    const claim = await civilServiceClient.retrieveClaimDetails(claimId, <AppRequest>req);
```
Comment (LeonardoPalmeiro) [inline src/main/routes/features/caseProgression/trialArrangements/finaliseTrialArrangementsController.ts:17]: we can use const claim = await getClaimById(claimId, req);

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by m-meulendijks-v1 — DISMISSED** [review]

File: src/main/views/features/dashboard/item-content.njk [hunk 3]
```diff
@@ -306,4 +306,26 @@
       </select>
     </div>
   {% endif %}
+
+  {% if content.type === ClaimSummaryType.LEAD_PARAGRAPH_WITH_NO_BOTTOM_MARGIN %}
```
[the remark this answers is not in the document]
Comment (hmcts-version1-ignacio) [reply src/main/views/features/dashboard/item-content.njk:300]: Yes, I've done that, and added "classes" to the builder like this:

addLeadParagraph(text: string, variables?: unknown, classes?: string) {
    const leadParagraphSection = ({
      type: ClaimSummaryType.LEAD_PARAGRAPH,
      data: {
        text: text,
        variables: variables,
        classes: classes,
      },
    });
    this._claimSummarySections.push(leadParagraphSection);
    return this;
  }

[the remark this answers is not in the document]
Comment (bhagyashreesharma90) [reply src/main/views/features/dashboard/item-content.njk]: > I think we dont need to introduce a new LEAD_PARAGRAPH_WITH_NO_BOTTOM_MARGIN but instead we can reuse LeadParagraph but pass in the classes through params: for eg:
> 
> ```
>   {% if content.type === ClaimSummaryType.LEAD_PARAGRAPH %}
>       {% set object = '' %}
>       {% if content.data.variables %}
>           {% set object = content.data.variables %}
>           <p class="govuk-body-l **{{ content.data.classes }}**">{{ t(content.data.text,object)}}</p>
>       {% else %}
>           <p class="govuk-body-l **{{ content.data.classes }}**">{{ t(content.data.text)}}</p>
>       {% endif %}
>   {% endif %}
> ```
> 
> wdyt, this way the component can be reused with other classes

Done

**Review by LeonardoPalmeiro — DISMISSED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by kurt-uk — APPROVED** [review]

**Pull request merged**